### ISSUE TITLE
Plugins Logging

### ISSUE THREAD

**Issue by tokejepsen** [pyblish/pyblish-base#39]
Fiddling with a validator now, wanted some feedback from the plugin so I just used a print statement. Shouldn't we implement a logging from the plugins?

**Comment by mottosso** [comment]
At the moment, there is a logging class-decorator which adds a `self.log` to affected classes. See [here](https://github.com/abstractfactory/publish/blob/master/publish/plugins/extract_model.py#L12)

``` python
# Example
import publish.lib

@publish.lib.log
class MyClass(...):
    def process(self):
        self.log.info("Hello world!")
```

At the moment, the logging isn't being initialised in Publish, I've relied on Maya to do this so far. I've had issues with logging and Maya in that when Publish initialises one then Maya logs twice. Do you know any way around that? 

Here is how I usually setup logging.

``` python
def get_formatter():
    formatter = logging.Formatter(
        '%(asctime)s - '
        '%(levelname)s - '
        '%(name)s - '
        '%(message)s',
        '%H:%M:%S')
    return formatter


def setup_log():
    log = logging.getLogger('publish')

    if log.handlers:
        return log.handlers[0]

    log.setLevel(logging.INFO)

    formatter = get_formatter()
    stream_handler = logging.StreamHandler()
    stream_handler.setFormatter(formatter)
    log.addHandler(stream_handler)

    return log
```

**Comment by tokejepsen** [comment]
I have little to no experience with logging. Really my first project with logging:)

I'll investigate a bit more.

**Comment by mottosso** [comment]
Ok, no worries. :) Just append the decorator to your validator, and use the `self.log` member. We can tailor logging afterwards, but this can remain the interface until we find something better.

**Comment by mottosso** [comment]
In the current version, there is some initialisation of logging happening in [`__init__.py`](https://github.com/abstractfactory/publish/blob/143c80010e69c78230658aec79460f4bdd6d1579/publish/__init__.py). If you notice in Maya, each logging entry happens twice because of this. If logging isn't initialised, then it won't produce any printing outside of Maya, and possibly not in other hosts either as I believe Maya is being particularly "smart" in this regard.

If anyone can think of a reason for this or how to solve it, feel free to have a look.

**Comment by mottosso** [comment]
This has been fixed in 1.0.7!
